File: process_images.py
import os
from PIL import Image, ImageDraw

folder = r"d:\gestionCamion\fleetguard\assets\images\markers"

for filename in os.listdir(folder):
    if not (filename.startswith("truck_") and filename.endswith(".png") and not "_fixed" in filename):
        continue
        
    filepath = os.path.join(folder, filename)
    print("Processing", filename)
    try:
        img = Image.open(filepath).convert("RGBA")
        
        # 1. Resize down to 256x256
        img.thumbnail((256, 256), Image.Resampling.LANCZOS)
        
        # 2. Convert to RGBA pixel data
        pixels = img.load()
        width, height = img.size
        
        # 3. Simple greedy algorithm: The fake checkerboard pixels are roughly (R~G~B > 195).
        # We will iterate and set matching pixels to transparent.
        # To avoid erasing the white of the truck cabin too much, we look strictly at the
        # specific grays and whites.
        
        for y in range(height):
            for x in range(width):
                r, g, b, a = pixels[x, y]
                # Light checkerboard
                if r > 190 and r < 210 and g > 190 and g < 210 and b > 190 and b < 210:
                    pixels[x, y] = (0, 0, 0, 0)
                # White checkerboard
                elif r > 240 and g > 240 and b > 240:
                    pixels[x, y] = (0, 0, 0, 0)
                    
        # 4. Save
        new_filename = filename.replace(".png", "_fixed.png")
        img.save(os.path.join(folder, new_filename), "PNG")
        print("Saved", new_filename)
    except Exception as e:
        print("Error on", filename, e)

File: process_images.dart
import 'dart:io';
import 'package:image/image.dart' as img;

void main() {
  final dir = Directory('assets/images/markers');
  for (var file in dir.listSync()) {
    if (file is File && file.path.endsWith('.png') && !file.path.contains('_fixed.png')) {
      print('Processing ${file.path}');
      final raw = file.readAsBytesSync();
      final image = img.decodePng(raw);
      if (image == null) {
        print('Failed to decode ${file.path}');
        continue;
      }
      
      // Resize to 256
      final resized = img.copyResize(image, width: 256);
      
      // Remove White/Gray checkerboard
      for (final pixel in resized) {
        final r = pixel.r;
        final g = pixel.g;
        final b = pixel.b;
        if ((r > 190 && r < 215 && g > 190 && g < 215 && b > 190 && b < 215) ||
            (r > 240 && g > 240 && b > 240)) {
           pixel.a = 0;
        }
      }
      
      final outPath = file.path.replaceAll('.png', '_fixed.png');
      File(outPath).writeAsBytesSync(img.encodePng(resized));
      print('Saved $outPath');
    }
  }
}
